extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -28,76 +28,133 @@ public static function onBeforePageDisplay() {
+				'category' => $modelData['categoryId'],
+				'show' => $showLatest ? 'latest' : 'trending',
+				'itemCount' => $itemCount,
+			] ),
+			'loading' => wfMessage( 'embeddable-discussions-loading' )->plain()
+		];
+
+		// In mercury, discussions app is rendered client side in an Ember container
+		return $templateEngine->clearData()
+			->setData( $modelData )
+			->render( 'DiscussionThreadMobile.mustache' );
+	}
+
+	private static function renderDesktop( $modelData, $showLatest, $category, $columns, $templateEngine ) {
+		$modelData['requestData'] = json_encode( [
+			'category' => $category,
+			'columns' => $columns,
+			'columnsDetailsClass' => $columns === 2 ? 'embeddable-discussions-post-detail-columns' : '',
+			'showLatest' => $showLatest,
+			'upvoteRequestUrl' => $modelData['upvoteRequestUrl'],
+		] );
+
+		if ( $showLatest && $category ) {
+			$heading = wfMessage( 'embeddable-discussions-show-latest-in-category', $category )->plain();
+		} else if ( $showLatest ) {
+			$heading = wfMessage( 'embeddable-discussions-show-latest' )->plain();
+		} else if ( $category ) {
+			$heading = wfMessage( 'embeddable-discussions-show-trending-in-category', $category )->plain();
 		} else {
-			$modelData = ( new DiscussionsThreadModel( $wgCityId ) )->getData( $showLatest, $itemCount, $category );
-
-			$modelData['requestData'] = json_encode( [
-				'category' => $category,
-				'columns' => $columns,
-				'columnsDetailsClass' => $columns === 2 ? 'embeddable-discussions-post-detail-columns' : '',
-				'showLatest' => $showLatest,
-				'upvoteRequestUrl' => $modelData['upvoteRequestUrl'],
-			] );
-
-			if ( $showLatest && $category ) {
-				$heading = wfMessage( 'embeddable-discussions-show-latest-in-category', $category )->plain();
-			} else if ( $showLatest ) {
-				$heading = wfMessage( 'embeddable-discussions-show-latest' )->plain();
-			} else if ( $category ) {
-				$heading = wfMessage( 'embeddable-discussions-show-trending-in-category', $category )->plain();
-			} else {
-				$heading = wfMessage( 'embeddable-discussions-show-trending' )->plain();
-			}
+			$heading = wfMessage( 'embeddable-discussions-show-trending' )->plain();
+		}
+
+		$modelData['columnsWrapperClass'] = $columns === 2 ? 'embeddable-discussions-threads-columns' : '';
+		$modelData['heading'] = $heading;
+		$modelData['showAll'] = wfMessage( 'embeddable-discussions-show-all' )->plain();
+		$modelData['loading'] = wfMessage( 'embeddable-discussions-loading' )->plain();
+
+		return $templateEngine->clearData()
+			->setData( $modelData )
+			->render( 'DiscussionThreadDesktop.mustache' );
+
+	}
 
-			$modelData['columnsWrapperClass'] = $columns === 2 ? 'embeddable-discussions-threads-columns' : '';
-			$modelData['heading'] = $heading;
-			$modelData['showAll'] = wfMessage( 'embeddable-discussions-show-all' )->plain();
-			$modelData['loading'] = wfMessage( 'embeddable-discussions-loading' )->plain();
+	public static function render( $input, array $args ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "$input". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)